### ISSUE TITLE
App chokes on some unicode characters

### ISSUE THREAD

**Issue by joxl** [editxt/editxt#10]
Discovered this while trying to paste a chat log (containing emoji) into EditXT.

To reproduce:

``` bash
echo -e '\xf0\x9f\x98\x92' > unamused.txt
xt unamused.txt
```
**Observe**: EditXT Error log document opens with the following error:
```
ERROR editxt - *** Python exception discarded!
Traceback (most recent call last):
  File "/Users/Shared/Applications/EditXT.app/Contents/Resources/lib/python3.3/editxt/platform/mac/views/util.py", line 29, in wrapper
  File "/Users/Shared/Applications/EditXT.app/Contents/Resources/lib/python3.3/editxt/controls/linenumberview.py", line 155, in drawHashMarksAndLabelsInRect_
IndexError: string index out of range
```
**Expected**: `unamused.txt` file to open in EditXT.app and I'd see a nice little unamused emoji: ﻿﻿:unamused:.

At this point the entire app state is pretty bad, best just kill it and start over.

NOTE: I get the same result when opening the same file (`unamused.txt`) from the `File --> Open...` menu.

Not all such symbols create this bug however. The following works just fine:

``` bash
echo -e '\xe2\x80\x9chi!\xe2\x80\x9d' > unicode.txt
xt unicode.txt
```
**Observe**: (expected) `unicode.txt` opens in EditXT.app showing the contents: `“hi!”`

**Comment by millerdev** [comment]
Well this is disturbing:
```
>>> text = textview.string()
>>> text.length()
3
>>> len(text)
2
>>> text
'😒\n'
```
which is the same as `"\U0001f612\n"`

**Comment by millerdev** [comment]
Found it! Here's [the problem](http://www.objc.io/issue-9/unicode.html):

> an NSString object actually represents an array of UTF-16-encoded code units. Accordingly, the length method returns the number of code units (not characters) in the string.

**Comment by millerdev** [comment]
Fixed by 54127b559ab86ff0bd677ba04e1cd9f2baed1838
